code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -193,6 +193,10 @@
 	ADD_TRAIT(src, TRAIT_NOFIRE_SPREAD, ROUNDSTART_TRAIT)
 	for(var/trait in weather_immunities)
 		ADD_TRAIT(src, trait, ROUNDSTART_TRAIT)
+	if (environment_smash & ENVIRONMENT_SMASH_RWALLS)
+		AddElement(/datum/element/wall_smasher, strength_flag = ENVIRONMENT_SMASH_RWALLS)
+	else if (environment_smash & ENVIRONMENT_SMASH_WALLS)
+		AddElement(/datum/element/wall_smasher)
```

Comment:
Is this necessary? Can we instead just pass through strength_flag always?

Reply:
While it's not frequently used as one and I could get away with it, `environment_smash` on simple mobs is a bitflag so I am worried that if I pass it directly then if someone set it to `ENVIRONMENT_SMASH_STRUCTURES | ENVIRONMENT_SMASH_WALLS` that is not the same value as `ENVIRONMENT_SMASH_WALLS`

That said maybe I should just refactor this not to be a bitflag because... it doesn't seem to benefit from being one?